src/tests/ftest/util/run_utils.py

```diff
@@ -551,19 +558,25 @@ def stop_processes(log, hosts, pattern, verbose=True, timeout=60, exclude=None,
     """
     processes_detected = NodeSet()
     processes_running = NodeSet()
-    command = f"/usr/bin/pgrep --list-full {pattern}"
+    search_command = f"/usr/bin/pgrep --list-full {pattern}"
     pattern_match = str(pattern)
     if exclude:
-        command = f"/usr/bin/ps xa | grep -E {pattern} | grep -vE {exclude}"
+        search_command = f"/usr/bin/ps xa | grep -E {pattern} | grep -vE {exclude}"
         pattern_match += " and doesn't match " + str(exclude)
+    elif full_command:
+        search_command = f"/usr/bin/pgrep --list-full --full -x {pattern}"
 
     # Search for any active processes
     log.debug("Searching for any processes on %s that match %s", hosts, pattern_match)
-    result = run_remote(log, hosts, command, verbose, timeout)
+    result = run_remote(log, hosts, search_command, verbose, timeout)
     if not result.passed_hosts:
         log.debug("No processes found on %s that match %s", result.failed_hosts, pattern_match)
         return processes_detected, processes_running
 
+    # Catch any attempt to kill process 1.
+    if "1" in re.findall(r"^(\d+)\s+", result.joined_stdout, re.MULTILINE):
+        raise ValueError(f"Attempting to kill process 1 as a match for {pattern}!")
```

Comment:
Just FYI if you wanted to avoid regex. But safer to just stick with what you have
```
if any(map(lambda line: line == '1', list_of_things_to_search))
```

Reply:
Wouldn't the `list_of_things_to_search` need to be just the pids from `result.joined_stdout` - which is what the regex provides?

Reply:
Yeah, it would need to be a list of just the pids. It's really a matter of dealing with a regex like `r"^(\d+)\s+"` or using lambda or list comprehension.
`"1" in ... r"^(\d+)\s+"` effectively means `"1" == line`